gems/aws-sdk-core/lib/aws-sdk-core/rest/request/body.rb

```diff
@@ -45,20 +49,58 @@ def build_body(params)
             params[@rules[:payload]]
           elsif @rules[:payload]
             params = params[@rules[:payload]]
+            if xml_builder? &&
+               @rules.shape.member?(@rules[:payload_member].location_name)
+              # serializing payload member for rest-xml is as follows:
+              # 1. Use the member locationName if the member value doesn't match the member's name (default)
+              # 2. Use the value of the locationName on the member's target if present
+              # 3. Use the shape name of the member's target
+              update_payload_location_name
+            end
             serialize(@rules[:payload_member], params) if params
           else
             params = body_params(params)
             serialize(@rules, params) unless params.empty?
           end
         end
 
+        def update_payload_location_name
+          @rules[:payload_member].location_name =
```

Comment:
So we're modifying the actual rules/client_api here?  It would be better to not modify these if we can.  

Instead, can we move this logic into the actual serialization logic?

Reply:
Yes, agreed. I'm going to look into having a keyword arg to pass this specific information down to serialization.